app/models/service_template_terraform_template.rb

```diff
@@ -37,6 +37,9 @@ def self.validate_config_info(info)
 
   def terraform_template(action)
     template_id = config_info.dig(action.downcase.to_sym, :configuration_script_payload_id)
+    if template_id.nil? && action.downcase == 'retirement'
+      template_id = config_info.dig(:provision, :configuration_script_payload_id)
```

Comment:
Is this saying that when doing retirement use the provisioning workflow?

Reply:
By default, for terraform ,  same terraform template files, vars,  but different action,  ie when 'Retirement' action -> run destroy

Reply:
@agrare,  with the changes, now the template_id will be available with Retirement options also, in ServiceTemplateTerraformTemplate,